hedera-node/src/main/java/com/hedera/services/contracts/execution/EvmTxProcessor.java

```diff
@@ -231,10 +231,6 @@ protected TransactionProcessingResult execute(Account sender, Address receiver,
 			process(messageFrameStack.peekFirst(), new HederaTracer());
 		}
 
-		if (initialFrame.getState() == MessageFrame.State.COMPLETED_SUCCESS && !isStatic) {
```

Comment:
Afaict this block has no use; the `process()` call above for this frame will have **already** called `commit()` if the frame completed successfully.